addons/custom-menu-bar/username.js

```diff
@@ -0,0 +1,14 @@
+export default async function ({ addon }) {
+  if (await addon.auth.fetchIsLoggedIn()) {
+    const username = await addon.auth.fetchUsername();
+    addon.tab.redux.initialize(); // Start listening to Redux events
+    addon.tab.redux.addEventListener("statechanged", async (e) => {
+      if (!addon.settings.get("compact-username")) return; // Skip if the option is disabled
+      if (e.detail.action.type !== "scratch-gui/menus/OPEN_MENU" || e.detail.action.menu !== "accountMenu") return; // Skip if another Redux event occurs
```

Comment:
So there's a total of 3 possible Redux events? I don't get what this is supposed to mean.

Reply:
The first is the event that's triggered when any menu is open and the second is which menu was just opened. I think not returning early here makes the code easier to understand.

Reply:
Huh, looks like I misread. These check for inequality, not equality. Sorry